STACpopulator/populator_base.py

```diff
@@ -112,8 +121,12 @@ def create_stac_collection(self) -> None:
         collection = pystac.Collection(**self._collection_info)
 
         collection.add_links(self._ingest_pipeline.links)
+        collection_data = collection.to_dict()
+        self.publish_stac_collection(collection_data)
+        return collection_data
 
-        post_stac_collection(self.stac_host, collection.to_dict(), self.update)
+    def publish_stac_collection(self, collection_data: MutableMapping[str, Any]) -> None:
```

Comment:
The test shows how it can be useful to decouple the logic. 
It is possible this way for an implementation to do something else with the created STAC Collection data before publishing it, such as adding more links, assets or metadata only relevant to the collection and not the items.

Reply:
OK